### PR TITLE
Implement template library and support blockly templates

### PR DESCRIPTION
**Pull request by janosbrodbeck**
**Based on #49 and #50**
This PR is implements my current **idea** how to implement pre-defined templates in the backend. Seeding templates into the database is not quite so trivial (when to update? What, if the template has been deleted? Seed it again? Etc.), so I came up with a simpler solution, which should fit our current needs. But is a bit more work for the frontend I guess.

* Adds an endpoint, which returns a list of pre-defined templates from the backend. They are not assignable to robot types, but instead can be "browsed" (frontend part) when the user wants to create a new template. The user could then look what templates are in the library and could use it as a starting point for their template. The List of templates is basically a normal Template structure wise. We can have a call about this idea as well to talk about some details and whether it is feasible.
Nonetheless: This results in way less problems when templates change, are deleted or added → less headache for the backend.
* A template file `.template` consists of a JSON in the first line with metadata, the rest of the file is the template itself
* Added a test, which tests if all templates can be loaded
* Fixes a bug in the generation template base schema with the `schema_extra` attribute

### CODE REVIEW

File: src/app/templates/getter.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import json
+from pathlib import Path
+from typing import List
+
+from app.schemas.generation_template import LibraryTemplate
+
+__path_to_testdata = Path(__file__).parent
+
+
+def __get_templates_in_path():
```
Comment (dolevi-tub) [inline src/app/templates/getter.py:11]:
```suggestion
def __get_templates_in_path() -> List[bytes]:
```

File: src/app/templates/getter.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+
+def get_count_template_files() -> int:
+    return len(__get_templates_in_path())
+
+
+def get_library_templates() -> List[LibraryTemplate]:
+    library_templates = []
+    for template in __get_templates_in_path():
+        f = open(template, "r")
```
Comment (dolevi-tub) [inline src/app/templates/getter.py:27]:
```suggestion
        with open(template, "r") as file:
        # ...
```

**Review by dolevi-tub — CHANGES_REQUESTED** [review]

File: src/app/templates/getter.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (janosbrodbeck) [reply src/app/templates/getter.py]: Addressed

File: src/app/templates/getter.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import os
+import json
+from pathlib import Path
+from typing import List
+
+from app.schemas.generation_template import LibraryTemplate
+
+__path_to_testdata = Path(__file__).parent
```
[the remark this answers is not in the document]
Comment (janosbrodbeck) [reply src/app/templates/getter.py]: Addressed, good old copypasta

File: src/app/templates/getter.py (hunk 1 above)
Comment (janosbrodbeck) [reply src/app/templates/getter.py]: Addressed

File: src/app/templates/getter.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+
+def __get_templates_in_path():
+    templates = []
+    for path in os.listdir(__path_to_testdata):
+        filepath = os.path.join(__path_to_testdata, path)
+        if os.path.isfile(filepath) and path.split(".")[-1] == "template":
```
[the remark this answers is not in the document]
Comment (janosbrodbeck) [reply src/app/templates/getter.py]: Addressed

File: src/app/templates/getter.py (hunk 2 above)
Comment (janosbrodbeck) [reply src/app/templates/getter.py]: Addressed

File: src/app/templates/getter.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            # First line contains JSON string with metadata
+            json_obj = json.loads(content[0])
+
+            # Remaining lines are the template
+            content = content[1:]
+            json_obj["content"] = "".join(content)
+
+            library_templates.append(LibraryTemplate(**json_obj))
+        except Exception as e: # noqa (PEP E722)
+            print(f"Template file '{os.path.basename(template)}' cannot be loaded: {e}")
```
[the remark this answers is not in the document]
Comment (janosbrodbeck) [reply src/app/templates/getter.py]: Added the comment, thanks :)

File: src/app/templates/getter.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (janosbrodbeck) [reply src/app/templates/getter.py]: Placed now a `# noqa` here ¯\_(ツ)_/¯

**Comment by janosbrodbeck** [comment]
I am thinking about adding a `version` field as well

**Comment by janosbrodbeck** [comment]
Added support for blockly templates and added a very basic example blockly template. Sorry for including a massive library like `beautifulsoup` but the built-in XML parser from python outputs crappy ""prettified"" code.  It's not completely how I want it right now, but it's good enough if someone has a quick look at it.

**Comment by janosbrodbeck** [comment]
Implemented support for blockly templates. Blockly templates arerettified again, as the resulted generated code is not well formatted. Also added an example blockly template for the niryo one (new stack).

**Review by dolevi-tub — APPROVED** [review]
Nothing to note :+1:

**Pull request merged**